app/main/views/index.py

```diff
@@ -1,3 +1,5 @@
+from urllib.parse import urlencode
+
```

Comment:
May be easier to look at this whole file.

Clients start on landing, the `filename` arg is passed to the second view (`download_document` method) and then to the document API to download the file.